agent/structs/check_type.go

```diff
@@ -48,32 +48,47 @@ func (c *CheckType) Valid() bool {
 	return c.IsTTL() || c.IsMonitor() || c.IsHTTP() || c.IsTCP() || c.IsDocker()
 }
 
+// Empty checks if the CheckType has no fields defined. Empty checks parsed from json configs are filtered out
+func (c *CheckType) Empty() bool {
+	return c.Script == "" && c.ScriptArgs == nil && c.HTTP == "" && (c.Header == nil || len(c.Header) == 0) && c.Method == "" && c.TCP == "" && c.DockerContainerID == "" && c.Shell == "" && c.TTL == 0 && c.Interval == 0
+}
+
 // IsScript checks if this is a check that execs some kind of script.
 func (c *CheckType) IsScript() bool {
 	return c.Script != "" || len(c.ScriptArgs) > 0
 }
 
 // IsTTL checks if this is a TTL type
 func (c *CheckType) IsTTL() bool {
-	return c.TTL != 0
+	return c.TTL > 0
```

Comment:
We were allowing negative TTLs and intervals for some reason. I fixed that as part of this change. 

For TTL checks there are no other required fields, so this is the only way to distinguish between valid vs empty (0 is the default value when the struct definition for a service is parsed out of json).